openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -412,6 +413,7 @@ def post(self, request):
                 # Delete OAuth tokens associated with the user.
                 retire_dop_oauth2_models(request.user)
                 retire_dot_oauth2_models(request.user)
+                AccountRecovery.retire_recovery_email(request.user.id)
```

Comment:
I'll note here that this occurs in the first step of the user retirement process before the cool-off period. During that time, users have been known to change their mind *or* contact us stating that retirement was a mistake. When those accounts are restored, the original username and original email address can be safely retrieved and restored. 

The way this is written, the secondary email address will *not* be able to be recovered for the un-retired accounts. If that meets product requirements, then it's fine. If the secondary email address needs to be recoverable in those cases, it'd need to be stored in the retirement status table along with the original username/email address.